src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/SurrogateSelector.cs

```diff
@@ -67,7 +67,7 @@ public virtual void ChainSelector(ISurrogateSelector selector)
             // Verify that we don't try and add ourself twice.
             if (selector == this)
             {
-                throw new SerializationException(SR.Serialization_DuplicateSelector);
+                throw new SerializationException(SR.Format(SR.Serialization_DuplicateSelector, this, selector));
```

Comment:
I think this is supposed to be something like

```C#
throw new SerializationException(
    SR.Format(
        SR.Serialization_DuplicateSelector,
        nameof(SurrogateSelector),
        selector.GetType().Name));
```

But it's hard to say.  @joperezr or @AlexGhiondea: thoughts?

Reply:
Yeah, let's change it to Jeremy's suggestion.

Reply:
Although... this error message seems wrong.  It seems like this is just a part of the next HasCycle case (which also doesn't make sense to me... we throw if !HasCycle rather than if HasCycle)?

Reply:
It looks HasCycle has a bug; it returns false when it should return true, and vice versa.  Then this only call site of it proves the saying "two wrongs make a right".

Reply:
For now, I suggest just changing this to:
```C#
if (selector == this)
{
    throw new ArgumentException(SR.Serialization_SurrogateCycle, nameof(selector));
}
```

Reply:
I agree that the message seems wrong, but it is the one NetFx uses (though it also doesn't format it it), so my suggestion was trying to keep parity with that.

Keeping the type as SerializationException and changing it to SR.Serialization_SurrogateCycle seems fine to me, though.  (But there's no spot for the parameter name in that exception type, so it would just be `new SerializationException(SR.Serialization_SurrogateCycle)`).

Reply:
let's go with new SerializationException(SR.Serialization_SurrogateCycle) seems a good compromise.